### PR TITLE
Bug Fixed : request confirmation before running keeps showing loader, when click cancel

### LINKED ISSUE #3640: Bug in Request confirmation before running query
### Environment
Development

### What is the expected behaviour?
When we toggled `Request confirmation before running query` to enable, and then clicked cancel on the pop-up window, the query should not run, and the button to run query should be enabled and should not have a loader 
### What is the current behaviour?
When we toggled `Request confirmation before running query` to enable, and then click cancel on the pop-up window, the  query was not running, but the button to run query keeps on showing a loader, and only goes after refreshing the page

### How to reproduce the issue?

1. Create a query or you can select any one of the earlier created queries.
2. Select the Advanced tab 
3. Toggled on/enable the Request confirmation before running the query option
4. Click on `Save & run` or `Create & run`
5. A pop-up will show up - "Do you want to run this query - <query name>", click on the cancel, a button that fired this pop-up, will keep on showing a loader

### PR DESCRIPTION
**Pull request by manishkushare**
Resolve #3640

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-2kbnc3q Bug in Request confirmation before running query](https://app.clickup.com/t/2kbnc3q)

**Comment by Nandinisaha13** [comment]
<html>
<body>
<google-sheets-html-origin>Test Cases

Test Cases | Steps | Result
-- | -- | --
Should not be able to show a loader on save & run query  while clicked on cancel. | Create an application. | Application gets created.
  | Click on add new query. | Different datasource is displayed.
  | Click on any datatsource. | Query page for datasource is displayed.
  | Go to advanced. | Advanced page is displayed.
  | Toggle on request confirmation before running query. | The request for confirmation before running query is toggled on.
  | Click on create an run. | The pop up for confirmation is displayed.
  | Click on cancel. | There is no loader on create and run.
Should be able to show loader on create & run or save & run. | Create an application. | Application gets created.
  | Click on add new query. | Different datasource is displayed.
  | Click on any datatsource. | Query page for datasource is displayed.
  | Toggle off request confirmation before running query. | The request for confirmation before running query is toggled off.
  | Go to general. | General page for query is displayed.
  | Click on create & run or save & run. | The loader is displayed until it is created & run or saved & run.


</body>
</html>

File: frontend/src/Editor/QueryManager/QueryManager.jsx [hunk 1]
```diff
@@ -102,6 +102,7 @@ let QueryManager = class QueryManager extends React.Component {
           props.mode === 'edit'
             ? this.buttonConfig?.editMode?.shouldRunQuery ?? true
             : this.buttonConfig?.createMode?.shouldRunQuery ?? true,
+        showQueryConfirmation: props.showQueryConfirmation,
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/QueryManager/QueryManager.jsx:105]: This line can be removed. We're not using the state anywhere in the file

File: frontend/src/Editor/QueryManager/QueryManager.jsx [hunk 2]
```diff
@@ -259,7 +272,7 @@ let QueryManager = class QueryManager extends React.Component {
   };
 
   createOrUpdateDataQuery = () => {
-    const { appId, options, selectedDataSource, mode, queryName, shouldRunQuery } = this.state;
+    const { appId, options, selectedDataSource, mode, queryName, shouldRunQuery, showQueryConfirmation } = this.state;
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/QueryManager/QueryManager.jsx:275]:
```suggestion
    const { appId, options, selectedDataSource, mode, queryName, shouldRunQuery } = this.state;
```

File: frontend/src/Editor/QueryManager/QueryManager.jsx [hunk 3]
```diff
@@ -398,6 +411,7 @@ let QueryManager = class QueryManager extends React.Component {
       previewLoading,
       queryPreviewData,
       dataSourceMeta,
+      showQueryConfirmation,
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/QueryManager/QueryManager.jsx:414]:
```suggestion
```

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@manishkushare Minor changes, we're not using the `showQueryConfirmation` in the file other than checking it in `componentWillReceiveProps`. So, we can make use of props and the state can be removed

**Comment by manishkushare** [comment]
@kavinvenkatachalam , updated the PR with suggested changes, please take a look at it, whenever possible.

**Review by kavinvenkatachalam — APPROVED** [review]

**Pull request merged**